crates/api_crud/src/site/create.rs

```diff
@@ -157,3 +141,148 @@ impl PerformCrud for CreateSite {
     })
   }
 }
+
+fn validate_create_payload(
+  is_site_setup: bool,
+  site_regex: Option<String>,
+  create_site: &CreateSite,
+) -> LemmyResult<()> {
+  // Make sure the site hasn't already been set up...
+  if is_site_setup {
+    return Err(LemmyError::from_message("site_already_exists"));
+  };
+
+  // Check that the slur regex compiles, and returns the regex if valid...
+  let slur_regex = build_and_check_regex(&site_regex.as_deref())?;
+
+  // Validate the site name...
+  site_name_length_check(&create_site.name)?;
+  check_slurs(&create_site.name, &slur_regex)?;
+
+  // If provided, validate the site description...
+  if let Some(desc) = &create_site.description {
+    site_description_length_check(desc)?;
+    check_slurs_opt(&create_site.description, &slur_regex)?;
+  }
+
+  // Ensure that the sidebar has fewer than the max num characters...
+  is_valid_body_field(&create_site.sidebar)
+}
+
+#[cfg(test)]
+mod tests {
+  use crate::site::create::validate_create_payload;
+  use lemmy_api_common::site::CreateSite;
+
+  #[test]
+  fn test_validate_create() {
+    fn create_payload(
+      site_name: String,
+      site_description: Option<String>,
+      site_sidebar: Option<String>,
+    ) -> CreateSite {
+      CreateSite {
+        name: site_name,
+        sidebar: site_sidebar,
+        description: site_description,
+        icon: None,
+        banner: None,
+        enable_downvotes: None,
+        enable_nsfw: None,
+        community_creation_admin_only: None,
+        require_email_verification: None,
+        application_question: None,
+        private_instance: None,
+        default_theme: None,
+        default_post_listing_type: None,
+        legal_information: None,
+        application_email_admins: None,
+        hide_modlog_mod_names: None,
+        discussion_languages: None,
+        slur_filter_regex: None,
+        actor_name_max_length: None,
+        rate_limit_message: None,
+        rate_limit_message_per_second: None,
+        rate_limit_post: None,
+        rate_limit_post_per_second: None,
+        rate_limit_register: None,
+        rate_limit_register_per_second: None,
+        rate_limit_image: None,
+        rate_limit_image_per_second: None,
+        rate_limit_comment: None,
+        rate_limit_comment_per_second: None,
+        rate_limit_search: None,
+        rate_limit_search_per_second: None,
+        federation_enabled: None,
+        federation_debug: None,
+        federation_worker_count: None,
+        captcha_enabled: None,
+        captcha_difficulty: None,
+        allowed_instances: None,
+        blocked_instances: None,
+        taglines: None,
+        registration_mode: None,
+        auth: Default::default(),
+      }
+    }
+
+    let invalid_payloads = [(
```

Comment:
I didn't add cases for functions from `validation.rs` because the unit tests in that file should cover those cases. Let me know if you think otherwise, and I can add more test cases here!